ethsigner/signer/hashicorp/src/main/java/tech/pegasys/ethsigner/signer/hashicorp/HashicorpVaultKVEngineClient.java

```diff
@@ -0,0 +1,149 @@
+      final HttpClient httpClient = vertx.createHttpClient(getHttpClientOptions(hashicorpConfig));
+      final String jsonResponse =
+          getVaultResponse(
+              httpClient,
+              hashicorpConfig.getAuthFilePath(),
+              requestURI,
+              hashicorpConfig.getTimeout());
+      return new JsonObject(jsonResponse);
+
+    } catch (final DecodeException de) {
+      throw new TransactionSignerInitializationException(ERROR_INVALID_JSON, de);
+    } finally {
+      if (vertx != null) {
+        vertx.close();
+      }
+    }
+  }
+
+  private HttpClientOptions getHttpClientOptions(final HashicorpConfig hashicorpConfig) {
+    final HttpClientOptions httpClientOptions =
+        new HttpClientOptions()
+            .setDefaultHost(hashicorpConfig.getHost())
+            .setDefaultPort(hashicorpConfig.getPort())
+            .setSsl(hashicorpConfig.isTlsEnabled());
+
+    if (hashicorpConfig.isTlsEnabled()) {
+      hashicorpConfig
+          .getTlsKnownServerFile()
+          .ifPresent(
+              knownServerFile ->
+                  httpClientOptions.setTrustOptions(whitelistServers(knownServerFile)));
+    }
+
+    return httpClientOptions;
+  }
+
+  private String getVaultResponse(
+      final HttpClient httpClient,
+      final Path authFilePath,
+      final String requestURI,
+      final long timeout) {
+
+    final CompletableFuture<String> future = new CompletableFuture<>();
+    final HttpClientRequest request =
+        httpClient.request(
+            HttpMethod.GET,
+            requestURI,
+            rh ->
+                rh.bodyHandler(
+                    bh -> {
+                      if (rh.statusCode() == 200) {
+                        future.complete(bh.toString());
+                      } else {
+                        future.completeExceptionally(
+                            new Exception(
+                                String.format(
+                                    "Hashicorp vault responded with status code {%d}",
+                                    rh.statusCode())));
+                      }
+                    }));
+    request.headers().set("X-Vault-Token", readTokenFromFile(authFilePath));
+    request.setChunked(false);
+    request.end();
+    return getResponse(future, timeout);
+  }
+
+  private String readTokenFromFile(final Path path) {
+    try (final Stream<String> stream = Files.lines(path)) {
+      return stream
+          .findFirst()
+          .orElseThrow(
+              () ->
+                  new TransactionSignerInitializationException(
+                      String.format(AUTH_FILE_ERROR_MSG_FMT, path.toString())));
+    } catch (final IOException e) {
+      LOG.error(e);
+      throw new TransactionSignerInitializationException(
+          String.format(AUTH_FILE_ERROR_MSG_FMT, path.toString()));
+    }
+  }
+
+  private String getResponse(final CompletableFuture<String> future, final long timeout) {
+    final String response;
```

Comment:
nit: no need to define this upfront, we can just do a return in the try block

Reply:
done.